library/gcm.c

```diff
@@ -332,28 +332,53 @@ int mbedtls_gcm_starts( mbedtls_gcm_context *ctx,
     return( 0 );
 }
 
-
+/**
+ * mbedtls_gcm_context::buf contains different data type, depending
```

Comment:
It's not “different data type”: it's always the partial state of the computation of the authentication tag. The breakdown is what partially processed input is at different stages.